test/test_evaluation.py

```diff
@@ -98,7 +98,9 @@ def test_evaluation(str_expr: str, str_expected: str, message=""):
         ),
     ],
 )
-@pytest.mark.skip
+@pytest.mark.skip(
+    reason="the right behaviour was broken since we start to use Symbol as singleton, to speedup comparisons."
+)
```

Comment:
Thanks!

Reply:
> the right behaviour was broken since we start to use Symbol as singleton, to speedup comparison

@mmatera I've been thinking about this. I suspect like a number of other things in this codebase, there was a fundamental flaw in thinking about Symbols and how bindings on Symbols work. The same sort of mess we see with M-Expressions and Elements (called oddly leaves), and how boxing and formatting should operate, along with the weirdly named InstanceableBuiltin. 

In other words, I don't think this is a simple thing that anyone will be able to fix by changing just the code in rewrite_apply_eval_step.

Reply:
I have the following idea: what about if, when we find an Element of the form ``Unevaluated[s]`` with ``s`` an Atom, we replace it with ``Sequence[s]``. Now, we have an Expression, and then, we can tag it without any problem.  Just before applying rules, all the sequences must be flattened anyway, so this adds an overhead just if there are elements tagged as ``Unevaluated``. If the rules fail to be applied, we go back to the expression with ``Sequence[s]`` on it, and then, depending on the value of the property ``unevaluated`` we can replace it back with ``Unevaluated[s]``.

Reply:
> I have the following idea:

Please, no. I have the following idea: learn more about how conventional interpreters work and make this code fit more into that framework. And then make the proposal if this isn't more simply handled in that framework. 

We may eventually do something like this. But I'd like do this starting from a framework that doesn't feel so homegrown and ignorant of common practice. 

Remember the days when string search and replace on MathML text for special strings like "->"  was proposed, because of a lack of understanding expression-driving formatting worked? I hope we can agree that the shotgun approach of search and replace on output strings isn't as good changing ToString for the Rule operator to output the unicode symbol when it is appropriate. 

Right now, this feel in that kind of category.

Reply:
If you want to write the fix outside of the Python code (i.e. "core")  such as in a file that gets autoloaded, e.g. a `.m` then have at it! 

Otherwise please wait.

Reply:
Actually, that could be a good solution. I am going to check if it is possible.